### PR TITLE
MAT-7484: Fix QDM ValueSet Suffix Adding/Changing Behavior

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-7484](https://jira.cms.gov/browse/MAT-7484)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/editor/valuesetApplier.ts [hunk 1]
```diff
@@ -122,61 +144,39 @@ const getValueSetTitleName = (vs) => {
...
 ): CodeChangeResult => {
-  const updatedCql = checkIfValueSetBeingEdited(vs, previousVs)
-    ? updateCql(cql, previousVs)
-    : cql;
-  const cqlArr: string[] = updatedCql.split("\n");
-  const parseResults: CqlResult = new CqlAntlr(updatedCql).parse();
+  const cqlArr: string[] = cql.split("\n");
+  const parseResults: CqlResult = new CqlAntlr(cql).parse();
   let valuesetChangeStatus: "success" | "info" | "danger" = "danger";
   let message: string = "The requested operation was unsuccessful";
```
Comment (adongare) [inline src/components/editMeasure/editor/valuesetApplier.ts]: value not being used, can be removed I think 
```suggestion
  let message: string;
```

File: src/components/editMeasure/editor/valuesetApplier.ts [hunk 2]
```diff
@@ -122,61 +144,39 @@ const getValueSetTitleName = (vs) => {
...
-  let vsExists: boolean = false;
+  let vsExactExists: boolean = false;
+  let vsSameTitleExist: boolean = false;
 
   // are there valuesets at all?
   if (parseResults?.valueSets?.length > 0) {
     parseResults.valueSets.forEach((valueSet) => {
```
Comment (adongare) [inline src/components/editMeasure/editor/valuesetApplier.ts]: Although this is somethign that exists already, I think we can improve it lil bit. If the match found, we don't need to continue iterating over rest of the list. foreach can't be broken. we could use ~parseResults.valueSets.some(..)~   for loop

File: src/components/editMeasure/editor/valuesetApplier.ts [hunk 3]
```diff
@@ -122,61 +144,39 @@ const getValueSetTitleName = (vs) => {
...
       const oldVsName = valueSet.name.replace(/["']/g, "");
       const oldUrl = valueSet.url.replace(/["']/g, "");
-      vsExists =
-        vsExists ||
-        (oldVsName === getValueSetTitleName(vs) && oldUrl === vs.oid) ||
+      vsExactExists =
+        vsExactExists ||
+        (oldVsName === getValueSetTitleName(vs) && oldUrl === vs.oid);
```
Comment (adongare) [inline src/components/editMeasure/editor/valuesetApplier.ts]:
```suggestion
      vsExactExists =
        oldVsName === getValueSetTitleName(vs) && oldUrl === vs.oid;
```

File: src/components/editMeasure/editor/valuesetApplier.ts [hunk 4]
```diff
@@ -122,61 +144,39 @@ const getValueSetTitleName = (vs) => {
...
+      vsSameTitleExist =
+        vsSameTitleExist ||
+        (oldVsName &&
+          extractValueSetNameAndSuffix(oldVsName)
+            .baseValueSetName.toLowerCase()
+            .replace(/\s/g, "") === vs.name.toLowerCase() &&
+          oldUrl === vs.oid) ||
```
Comment (adongare) [inline src/components/editMeasure/editor/valuesetApplier.ts]:
```suggestion
      vsSameTitleExist =
        (oldVsName &&
          extractValueSetNameAndSuffix(oldVsName)
            .baseValueSetName.toLowerCase()
            .replace(/\s/g, "") === vs.name.toLowerCase() &&
          oldUrl === vs.oid) ||
```

File: src/components/editMeasure/editor/valuesetApplier.ts (hunk 3 above)
Comment (ethankaplan) [reply src/components/editMeasure/editor/valuesetApplier.ts]: done

File: src/components/editMeasure/editor/valuesetApplier.ts (hunk 4 above)
Comment (ethankaplan) [reply src/components/editMeasure/editor/valuesetApplier.ts]: done

**Review by adongare — APPROVED** [review]

**Pull request merged**